plugins/module/create_kafka.py

```diff
@@ -0,0 +1,152 @@
+    - my_namespace.my_collection.my_doc_fragment_name
+
+author:
+    - Red Hat Developer
+'''
+
+EXAMPLES = r'''
+# Pass in a message
+- name: Test with a message
+  my_namespace.my_collection.create_kafka:
+    name: my-kafka
+
+ 
+'''
+
+RETURN = r'''
+# These are examples of possible return values, and in general should use other names for return values.
+original_message:
+    description: The original name param that was passed in.
+    type: str
+    returned: always
+    sample: 'hello world'
+message:
+    description: The output message that the test module generates.
+    type: str
+    returned: always
+    sample: 'goodbye'
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import rhoas_kafka_mgmt_sdk
+from rhoas_kafka_mgmt_sdk.api import default_api
+from rhoas_kafka_mgmt_sdk.model.cloud_provider_list import CloudProviderList
+from rhoas_kafka_mgmt_sdk.model.cloud_region_list import CloudRegionList
+from rhoas_kafka_mgmt_sdk.model.error import Error
+from rhoas_kafka_mgmt_sdk.model.kafka_request import KafkaRequest
+from rhoas_kafka_mgmt_sdk.model.kafka_request_list import KafkaRequestList
+from rhoas_kafka_mgmt_sdk.model.kafka_request_payload import KafkaRequestPayload
+from rhoas_kafka_mgmt_sdk.model.kafka_update_request import KafkaUpdateRequest
+from rhoas_kafka_mgmt_sdk.model.metrics_instant_query_list import MetricsInstantQueryList
+from rhoas_kafka_mgmt_sdk.model.metrics_range_query_list import MetricsRangeQueryList
+from rhoas_kafka_mgmt_sdk.model.supported_kafka_instance_types_list import SupportedKafkaInstanceTypesList
+from rhoas_kafka_mgmt_sdk.model.version_metadata import VersionMetadata
+import util.auth 
+
+configuration = rhoas_kafka_mgmt_sdk.Configuration(
+    host = "https://api.openshift.com"
+)
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict(
+        name=dict(type='str', required=True),
+    )
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # changed is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(
+        changed=False,
+        original_message='',
+        message=''
+    )
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    # if the user is working with this module in only check mode we do not
+    # want to make any changes to the environment, just return the current
+    # state with no modifications
+    if module.check_mode:
+        module.exit_json(**result)
+
+ 
+ 
+    token = util.auth.get_access_token()
```

Comment:
Token exchange works fine. Leaving up to you if we want to move this to Python SDK